### ISSUE TITLE
model zoom cannot be tracked

### ISSUE THREAD

**Issue by umzi2** [Supereeeee/EFEN#1]
Hello, you have done a very cool job, the architecture is stable and fast. But there is one serious problem: in all interpreters that use pytorch models, they will not be able to determine the degree of increase in the model since this does not affect the tensors in any way. I just decided to let you know about this problem.
2x
![изображение](https://github.com/Supereeeee/EFEN/assets/131161226/26a1cb2e-a1f0-4b78-be14-fa6b10406ae3)
3x
![изображение](https://github.com/Supereeeee/EFEN/assets/131161226/ad7501c3-b8b4-4d8d-aa06-a5134a9e493b)
4x
![изображение](https://github.com/Supereeeee/EFEN/assets/131161226/ce482c77-93bf-4c5f-96c3-e71f91d4b6b5)

**Comment by Supereeeee** [comment]
Sorry, I'm a little confused about your question. Do you mean that you can't determine exactly how the model parameters and computations have changed? If that's the case, you can modify the model in ./basicsr/archs/model_complexity.py file to change the overall number of channels and blocks in the model, or even the size of the BSConv kernel in a specific module, and then you can run ./basicsr/archs/model_complexity.py file to observe the variation of model parameters. If that's not what you mean, could you please be more detailed so I can explain it better for you?

**Comment by umzi2** [comment]
I apologize for not being able to explain. In general in all architectures most of the value changes are of dim, upscale, n_blocks type. They affect the number of tensors or their size in state_dict, due to which these models can be used in programs like chainner and similar ones. But you have a special case in your architecture when changing the upscale_factor parameter the tensors do not change in number or size. Because of this they cannot be used in many interpreters in their current form.
[example of how upscale is used in span](https://github.com/chaiNNer-org/spandrel/blob/main/libs/spandrel/spandrel/architectures/SPAN/__init__.py#L41)
In general, I'm not asking for a solution to the problem, it's a feature of the architecture structure, I just decided to inform you about this annoying problem.

**Comment by Supereeeee** [comment]
For general lightweight SR networks, 1conv+1pixelshuffle is often utilized for ×2, ×3 and x4 upsampling tasks, so that variations in channel number and feature size can be obviously observed when the upscale is changed. However, in this work, we adopted 2conv+1pixelshuffle for the ×2 and ×3 tasks, and repeated the x2 upsampling operation twice for the x4 task, so we set up the upsampling definition separately for each task, which makes all the defined upsampling layers appear simultaneously when printing the network structure. However, only the corresponding upampling layer is used when the upscale is determined. As for what you mentioned, we will make improvements in subsequent projects, and if time permits, we will also make corresponding improvements to the current work. Meanwhile, thank you very much for your kind reminder.
